lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/fpg/FPGrowthEmerging.java

```diff
@@ -147,33 +189,48 @@ public FPGrowthEmerging setCombinationsEnabled(boolean flag) {
             }
         }
 
-        // check the ratios of any itemsets we just marked
+        // Check the ratios of any itemsets we just marked
         FPGrowth inlierTree = new FPGrowth();
         List<ItemsetWithCount> matchingInlierCounts = inlierTree.getCounts(inliers,
                                                                            inlierCounts,
                                                                            ratioItemsToCheck,
                                                                            ratioSetsToCheck);
 
+        // Determine which of the ratio sets which satisfied the support threshold
+        // also satisfy all qualityMetrics.
         assert (matchingInlierCounts.size() == ratioSetsToCheck.size());
         for (int i = 0; i < matchingInlierCounts.size(); ++i) {
             ItemsetWithCount ic = matchingInlierCounts.get(i);
             ItemsetWithCount oc = ratioSetsToCheck.get(i);
 
-            double ratio = RiskRatio.compute(ic.getCount(),
-                                             oc.getCount(),
-                                             inliers.size(),
-                                             outliers.size());
+            boolean passThresholds = true;
+            double[] curAggregate = new double[]{
+                    oc.getCount(),
+                    oc.getCount() + ic.getCount()
+            };
+            for(int q = 0; q < qualityMetrics.size(); q++) {
+                double metric = qualityMetrics.get(q).value(curAggregate);
+                if (!(metric > thresholds.get(q))) {
+                    passThresholds = false;
+                }
+            }
 
-            if (ratio >= minRatio) {
-                ret.add(new FPGItemsetResult(oc.getCount() / (double) outliers.size(),
-                                          oc.getCount(),
-                                          ratio,
-                                          oc.getItems()));
+            if (passThresholds) {
+                double riskRatio = RiskRatio.compute(ic.getCount(),
```

Comment:
Why do we hardcode RiskRatio here? Since we're already generalizing fpgrowth to suport arbitrary quality metrics (with support still hardcoded) we should use the user-provided qualitymetric.